File: kotlin-typescript/karakum/plugins/convertSkippedGenerics.js
import ts from "typescript";

export default function (node, context, render) {
    if (
        ts.isTypeReferenceNode(node)
        && ts.isIdentifier(node.typeName)
        && node.typeName.text === "Visitor"
        && !node.typeArguments
    ) {
        return `${render(node.typeName)}<*, *>`
    }

    return null
}

File: kotlin-typescript/karakum/src/jsMain/kotlin/typescript/karakum/plugins/convertSkippedGenerics.kt
package typescript.karakum.plugins

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isIdentifier
import typescript.isTypeReferenceNode

val convertSkippedGenerics = createPlugin { node, _, render ->
    nullable {
        ensure(isTypeReferenceNode(node))

        val typeName = node.typeName
        ensure(isIdentifier(typeName))
        ensure(typeName.text == "Visitor")
        ensure(node.typeArguments == null)

        "${render(node.typeName)}<*, *>"
    }
}
